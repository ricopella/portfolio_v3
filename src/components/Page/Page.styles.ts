import styled from '@emotion/styled'
import { BREAKPOINTS, COLORS } from '../../styles/variables'

const StyledPage = styled.div`
  display: grid;
  align-items: center;
  grid-row-gap: 1rem;
  grid-template-columns: 1fr;
  grid-template-rows: max-content max-content;
  height: 100%;
  justify-content: center;
  margin: 5% 10% 0 10%;
  background-color: ${COLORS.header};
  width: 100%:

  @media (max-width: ${BREAKPOINTS.smRem}) {
    margin: 0 0;
  }
`

export default StyledPage
