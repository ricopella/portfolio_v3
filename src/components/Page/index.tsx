import PageHeading from '../PageHeading'
import React, { FC } from 'react'
import ScrollReveal from '../ScrollReveal'
import StyledPage from './Page.styles'
import { PageProps } from '../../types'

const Page: FC<PageProps> = ({ children, className, id, style, title }) => (
  <ScrollReveal>
    <StyledPage id={id} className={className} style={style}>
      <PageHeading title={title} />
      {children}
    </StyledPage>
  </ScrollReveal>
)

export default Page
