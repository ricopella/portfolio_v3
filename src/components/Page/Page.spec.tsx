import renderer from 'react-test-renderer'
import React from 'react'
import Page from './index'
import { render } from '@testing-library/react'

describe('<Page />', () => {
  const props = {
    className: 'foo',
    id: 'foo',
    style: { color: 'red' },
  }

  it('should match Page snapshot', () => {
    const internalTree = renderer
      .create(
        <Page {...props}>
          <div>Wu-Tang Clan</div>
        </Page>
      )
      .toJSON()

    expect(internalTree).toMatchSnapshot()
  })

  it('should render children', () => {
    const { getByText } = render(
      <Page {...props}>
        <div>Wu-Tang Clan</div>
      </Page>
    )

    expect(getByText(/Wu-Tang Clan/i)).toBeTruthy()
  })
})
