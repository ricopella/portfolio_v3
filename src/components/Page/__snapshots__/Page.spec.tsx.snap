// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page /> should match Page snapshot 1`] = `
<div
  data-sal="slide-up"
  data-sal-delay="200"
  data-sal-easing="ease"
>
  <div
    className="foo css-1p6xk4t"
    id="foo"
    style={
      Object {
        "color": "red",
      }
    }
  >
    <h1
      className="css-1cnsdig"
    >
      Wu-Tang Clan
    </h1>
    <div>
      Foo
    </div>
  </div>
</div>
`;
